src/managers/slidedownManager/SlidedownManager.ts

```diff
@@ -83,6 +87,24 @@ export class SlidedownManager {
       }
     } else {
       wasDismissed = DismissHelper.wasPromptOfTypeDismissed(DismissPrompt.NonPush);
+      const smsSubscribed = !!(await Database.getSMSProfile()).subscriptionId;
+      const emailSubscribed = !!(await Database.getEmailProfile()).subscriptionId;
+      const bothSubscribed = smsSubscribed && emailSubscribed;
+
+      if (smsSubscribed && (slidedownType === DelayedPromptType.Sms)) {
+        Log.info(new ExistingChannelError(DelayedPromptType.Sms));
+        return false;
+      }
+
+      if (emailSubscribed && (slidedownType === DelayedPromptType.Email)) {
+        Log.info(new ExistingChannelError(DelayedPromptType.Email));
+        return false;
+      }
+
+      if (bothSubscribed && (slidedownType === DelayedPromptType.SmsAndEmail)) {
+        Log.info(new ExistingChannelError(DelayedPromptType.SmsAndEmail));
+        return false;
+      }
```

Comment:
Notice we are checking for `isInUpdateMode` below all this new logic. Does calling `OneSignal.showSmsSlidedown({force: true});` still work to change the email or SMS?

Reply:
Seems like this would break the {force: true} behavior.
I think it make sense to allow force true to always show, even if we don't have an visual different for differences / autofill yet. This way we don't change the behavior later when we introduce that